### PR TITLE
[kube-prometheus-stack] support enableFeatures API

### LINKED ISSUE #823: [kube-prometheus-stack] Support EnableFeatures API
**Is your feature request related to a problem? Please describe.**
Prometheus Operator is adding the feature that allows users to enable the disabled features such as the in-memory storage for Exemplars. It's done via https://github.com/prometheus-operator/prometheus-operator/pull/3878, and merged. It would be great to see the Helm Chart support this.

**Describe the solution you'd like**
I want to be able to configure enableFeatures through `values.yaml` like this.
```yaml
prometheus;
  prometheusSpec:
    enableFeatures:
      # Enable the Exemplar in-memory storage
      - 'exemplar-storage'
```

**Additional Info**

- https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#prometheusspec
- See [the latest prometheuses.monitoring.coreos.com CRD](https://github.com/prometheus-operator/prometheus-operator/blob/master/example/prometheus-operator-crd/monitoring.coreos.com_prometheuses.yaml).

```yaml
              enableFeatures:
                description: Enable access to Prometheus disabled features. By default, no features are enabled. Enabling disabled features is entirely outside the scope of what the maintainers will support and by doing so, you accept that this behaviour may break at any time without notice. For more information see https://prometheus.io/docs/prometheus/latest/disabled_features/
                items:
                  type: string
                type: array
```

### PR DESCRIPTION
**Pull request by tonychoe**
#### What this PR does / why we need it:

Supports enableFeatures API of Prometheus Operator

#### Which issue this PR fixes
  - fixes #823

#### Special notes for your reviewer:
n/a

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/Chart.yaml
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:21]: new feature should raise minor version

**Review by monotek — CHANGES_REQUESTED** [review]

**Comment by tonychoe** [comment]
@monotek thanks, I changed the version. Please review and merge.

**Review by monotek — DISMISSED** [review]
Approved. 
But i can't merge as i'm not a maintainer of the chart.

**Review by bismarck — APPROVED** [review]
lgtm

**Pull request merged**
